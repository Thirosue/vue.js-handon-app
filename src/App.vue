<template>
  <div id="app">
    <section class="hero">
      <div class="hero-body">
        <div class="container">
          <h1 class="title">担当者一覧</h1>
        </div>
      </div>
    </section>
    <div class="container is-fullhd">
      <div class="notification">
        <div class="field sample-vue-app-input-firstName">
          <label for="firstName" class="label">名前</label>
          <div class="control">
            <input type="text" id="firstName" name="firstName" class="input" />
          </div>
        </div>
        <div class="field sample-vue-app-input-lastName">
          <label for="lastName" class="label">苗字</label>
          <div class="control">
            <input type="text" id="lastName" name="lastName" class="input" />
          </div>
        </div>
        <div class="field is-grouped is-grouped-right">
          <button @click="search" id="form-submit" type="submit" class="button is-link">Search</button>
        </div>
      </div>
      <hr />
      <section class="hero">
        <table class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
          <thead>
            <tr>
              <th>#</th>
              <th>ID</th>
              <th>名前</th>
              <th>苗字</th>
              <th>Email</th>
              <th>Tel</th>
            </tr>
          </thead>
          <tbody>
            <tr v-for="(staff, index) in results" :key="staff.id">
              <td>{{index+1}}</td>
              <td>
                <a href="#" class>{{staff.id}}</a>
              </td>
              <td>{{staff.firstName}}</td>
              <td>{{staff.lastName}}</td>
              <td>{{staff.email}}</td>
              <td>{{staff.tel}}</td>
            </tr>
          </tbody>
        </table>
      </section>
      {{results}}
    </div>
  </div>
</template>

<script>
export default {
  name: "app",
  data() {
    return {
      results: []
    };
  },
  methods: {
    search() {
      fetch("http://localhost:18081/admin/api/staff")
        .then(response => response.json())
        .then(myJson => {
          this.results = myJson.data;
          console.log(this.results);
        });
    }
  }
};
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
