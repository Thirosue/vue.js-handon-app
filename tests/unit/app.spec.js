import { shallowMount } from '@vue/test-utils'
import App from '@/App.vue'
import flushPromises from 'flush-promises'

//fetchのモック化
const results = require('./result.json')
window.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve({
    status: 200,
    ok: true,
    json: () => results
  })
)

describe('App.vue', () => {
  it('タイトルが表示されること', () => {
    const wrapper = shallowMount(App)
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('担当者一覧')
  })

  it('全件検索を実行したとき、検索結果が0件以上となること', async () => {
    const wrapper = shallowMount(App)

    //検索ボタンクリック
    wrapper.find('#form-submit').trigger('click')

    await flushPromises() //非同期処理を終了させる

    const resultList = wrapper.findAll('#result-list tr')
    expect(resultList.length).not.toEqual(0) //0 以上
    console.log(resultList.at(2).text())
  })

  it("snapshot", () => {
    const wrapper = shallowMount(App)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})