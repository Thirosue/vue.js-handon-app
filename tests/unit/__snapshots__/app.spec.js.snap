// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue snapshot 1`] = `
<div
  id="app"
>
  <section
    class="hero"
  >
    <div
      class="hero-body"
    >
      <div
        class="container"
      >
        <h1
          class="title"
        >
          担当者一覧
        </h1>
      </div>
    </div>
  </section>
   
  <div
    class="container is-fullhd"
  >
    <div
      class="notification"
    >
      <div
        class="field sample-vue-app-input-firstName"
      >
        <label
          class="label"
          for="firstName"
        >
          名前
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            id="firstName"
            name="firstName"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field sample-vue-app-input-lastName"
      >
        <label
          class="label"
          for="lastName"
        >
          苗字
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            id="lastName"
            name="lastName"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field is-grouped is-grouped-right"
      >
        <p
          class="control"
        >
          <button
            class="button is-link"
            id="form-submit"
            type="submit"
          >
            Search
          </button>
        </p>
         
        <p
          class="control"
        >
          <button
            class="button is-link"
            type="submit"
          >
            Shuffle
          </button>
        </p>
      </div>
    </div>
     
    <hr />
     
    <section
      class="hero"
      id="result-list"
    >
      <!---->
    </section>
    
    []
  
  </div>
</div>
`;
